code/modules/client/preferences_mysql.dm

```diff
@@ -57,7 +56,7 @@
 					UI_style='[UI_style]',
 					UI_style_color='[UI_style_color]',
 					UI_style_alpha='[UI_style_alpha]',
-					be_special='[be_special]',
+					be_role='[sql_sanitize_text(list2params(be_special))]',
```

Comment:
I assume this sql_sanitize guard is what's torpedoing the entry adding, and causing everything to be concatenated to a single entry. I'll create a proc to sql_sanitize the entries of a list, and use it here, so that we don't leave any opening for a "Bobby tables" attack